Add API endpoint to retry all failed bg jobs

tw4l opened the pull request:
Fixes #1395 

Adds new `POST /orgs/jobs/retryFailed` API endpoint to retry all failed background jobs.

Tested manually on local dev instance by setting invalid secret key for replica storage, starting a few crawls/uploads, waiting for replica jobs to fail, then calling new endpoint and verifying via `GET /orgs/jobs` that the failed jobs were restarted.

tw4l committed:
Add API endpoint to retry all failed bg jobs

ikreymer commented:
Nice! Could we also add an endpoint to retry all failed jobs, accessible only for superadmin? Perhaps `/api/orgs/all/retryAllFailed`?

tw4l commented:
> Nice! Could we also add an endpoint to retry all failed jobs, accessible only for superadmin? Perhaps `/api/orgs/all/retryAllFailed`?

Yeah, good call! Added `POST /orgs/all/jobs/retryFailed` to make the superuser `all` endpoint consistent with the other ones we have. Tested and working well! Also added a missing `oid` query to the org-specific endpoint.

tw4l committed:
Add POST /orgs/all/jobs/retryFailed endpoint

tw4l committed:
Add nightly test to check that superusers can't call retry all bg jobs

tw4l committed:
Nightly test fixups

tw4l commented:
Added and tested a nightly test to ensure that non-superusers get a 403 when calling `POST /orgs/all/jobs/retryFailed`

tw4l commented:
@ikreymer could use your eyes when you have a moment
